pipeline-runner/Dockerfile

```diff
@@ -105,11 +105,14 @@ FROM common AS dev
 # also install watchdog to automatically restart
 # when source files change
 RUN pip install -U jedi radian PyYAML watchdog[watchmedo]
+RUN apt update && apt -y install git
+RUN pip install memory_profiler
```

Comment:
do we need these changes?

Reply:
It only installs this in the development build. And given that we'll be working with big datasets, it's probably not a bad idea to standardize the memory profiling.